### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by lookatmyusername**

### CODE REVIEW

File: precheck/cmd/qualityChecks/qualityChecks.go [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+}
+
+func printErr(str string) {
+	default_message = `{"error": "` + str + `"}`
+	fmt.Print(default_message)
+}
+
+func retrieveMetrics(repositoryURL string, client HTTPClient, cfg sfdccfg) (*Record, error) {
+
+	api := `/services/data/v51.0/queryAll/?q=SELECT+Unit_Test_Coverage__c,Security_Critical_Issue_Count__c,Security_High_Issue_Count__c,Security_Medium_Issue_Count__c,Security_Low_Issue_Count__c+from+Quality_Metrics_SonarQube__c+Where+Quality_Metrics_SonarQube_Name__c+='{{repo_url}}'`
```
Comment (cx-lcp-cd-demo) [inline precheck/cmd/qualityChecks/qualityChecks.go:158]: pls externalise this

File: precheck/cmd/qualityChecks/qualityChecks.go [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+}
+
+func retreiveToken(uname, pwd, clientId, clientSecret string, client HTTPClient) (*sfdccfg, error) {
+	data := url.Values{}
+	data.Set("grant_type", "password")
+	data.Set("username", uname)
+	data.Set("password", pwd)
+	data.Set("client_id", clientId)
+	data.Set("client_secret", clientSecret)
+	request, err := http.NewRequest(http.MethodPost, "https://test.salesforce.com/services/oauth2/token", strings.NewReader(data.Encode()))
```
Comment (cx-lcp-cd-demo) [inline precheck/cmd/qualityChecks/qualityChecks.go:95]: pls externalise

File: precheck/cmd/csdlCheck/checkInCSDL.go [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+// Arguements that are needed to pass:
+// 1 - token
+// 2 - host
+// 3 - repoName
+// 4 - espSysId
+
+func validator(client HTTPClient, token, host, repoName, espSysId string) (csdlOutput, error) {
+
+	endpoint := host + "/services/apexrest/ThreatIndexQualifier/v1.0/scanThreatIndex"
```
Comment (cx-lcp-cd-demo) [inline precheck/cmd/csdlCheck/checkInCSDL.go:122]: pls externalise

File: precheck/cmd/csdlCheck/checkInCSDL.go [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+	Do(req *http.Request) (*http.Response, error)
+}
+
+// Arguements that are needed to pass:
+// 1 - grant type
+// 2 - client id
+// 3 - client_secret
+// 4 - user name
+// 5 - password
+// 6 - token URL
```
Comment (cx-lcp-cd-demo) [inline precheck/cmd/csdlCheck/checkInCSDL.go:75]: as these are specific to this repo, do we need to pass them from every pipeline?

**Review by cx-lcp-cd-demo — CHANGES_REQUESTED** [review]

File: precheck/cmd/qualityChecks/qualityChecks.go [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+		return nil, err
+	}
+	var cfg sfdccfg
+	cfg.Token = tr.Token_type + " " + tr.Access_token
+	cfg.InstanceUrl = tr.Instance_url
+	return &cfg, nil
+}
+
+func getRespository(projectId, token string, client HTTPClient) (string, bool) {
+	endpoint := "https://cylon-api.cisco.com/middleware/api/project/" + projectId
```
Comment (cx-lcp-cd-demo) [inline precheck/cmd/qualityChecks/qualityChecks.go:131]: pls externalise

**Review by rama2723 — APPROVED** [review]

**Pull request merged**